rust/template/tcp_channel/receiver.rs

```diff
@@ -46,6 +88,10 @@ where
     /// the system will assign a port that is free. To retrieve this
     /// assigned port (in the form of the full `SocketAddr`), use the
     /// `addr` method.
+    ///
+    /// Note that the receiver may require some time after creation
```

Comment:
Why is that? You will have called `bind` by the time this method returns, so it should be ok for `TcpSenders` to connect. The connections will be processed once the listener thread calls `accept`. I might be wrong.

Reply:
Based on my observations my conclusion was that until `accept` is issued you'd still get a connection refused when trying to connect. The mere act of `bind`ing only prepares the socket and "reserves" a port. Or so that was my conclusion. I'll verify tomorrow.

Reply:
From `TcpListener` documentation:

```
After creating a TcpListener by binding it to a socket address, it listens for incoming TCP connections. These can be accepted by calling accept or by iterating over the Incoming iterator returned by incoming.
```

What I think this means is that `TcpListener.bind()` internally calls `listen()` from `sys/socket.h`, which in turn means that the socket should already be queuing incoming connections.

BTW, I think there is nothing wrong with `TcpSender` retrying regardless of this; just trying to figure out what's going on...

Reply:
Yeah, I tried it out and it works as you suspected it would. I agree that eventually we may need a `retry_connect` but I guess for now we don't.